lib/core/src/irods_random.cpp

```diff
@@ -1,7 +1,35 @@
+#include "irods/irods_random.hpp"
+
+#include "irods/irods_exception.hpp"
+#include "irods/rodsErrorTable.h"
+
 #include <boost/random.hpp>
+#include <fmt/format.h>
 #include <openssl/rand.h>
-#include "irods/irods_random.hpp"
+
+#include <cctype>
 #include <ctime>
+#include <random>
+#include <string>
+
+namespace
+{
+    auto generate_random_alphanumeric_character() -> char
+    {
+        constexpr char min_char = '0';
+        constexpr char max_char = 'z';
```

Comment:
is this ascii?   if so... there are other things between 0 and z...

Reply:
Yes, see `std::isalnum` check below.

Reply:
while loop.   sorry.   as you were.